core/src/main/scala/cats/syntax/either.scala

```diff
@@ -63,6 +63,11 @@ final class EitherOps[A, B](val eab: Either[A, B]) extends AnyVal {
     case Right(b) => if (f(b)) eab else Left(onFailure)
   }
 
+  def ensureOr[AA >: A](onFailure: B => AA)(f: B => Boolean): Either[AA, B] = eab match {
```

Comment:
It looks like this is following a precedent, but I find myself wondering if this `Either` syntax should really be providing syntax for methods that should already be available via type class syntax (in this case `MonadError`). Let's table that for a separate issue though.